File: rxls-serve/rxls-common/src/main/java/cn/rx/common/enums/RoleEnum.java
package cn.rx.common.enums;

/**
 * 基础角色权限值、id枚举
 */
public enum RoleEnum {
    /**
     * 基础管理层角色权限值
     */
    ADMIN_VALUE(999),
    /**
     * 超级管理id
     */
    SUPER_ADMIN(1),
    /**
     * 管理层角色id
     */
    ADMIN(2),
    /**
     * 用户角色id
     */
    USER(3);
    /**
     * 权限值
     */
    public final Integer value;

    /**
     * 是否是管理员角色
     * @param id
     * @return
     */
    public static boolean isAdmin(Integer id){
        return ADMIN.value.equals(id)  || SUPER_ADMIN.value.equals(id);
    }

    RoleEnum(Integer value) {
        this.value = value;
    }
}

File: rxls-admin/src/config/enum/RoleEnum.ts
/**
 * 状态码枚举
 */
const RoleEnum = {
  /**
   * 管理员（大于等于此值为管理员）
   */
  ADMIN_VALUE: 100,

  /**
   * 基础角色普通用户ID
   */
  USER: 3,
  /**
   * 基础角色管理员ID
   */
  ADMIN: 2,

  /**
   * 基础角色超级管理员ID
   */
  SUPER_ADMIN: 1,
}

export { RoleEnum }
